libs/defi/shared/src/components/Activities/constants.tsx

```diff
@@ -0,0 +1,522 @@
+import {
+  BadgeIcon,
+  SwapTokensIcon,
+  TokenIcon,
+} from '@origin/shared/components';
+import { getTokenById } from '@origin/shared/contracts';
+import {
+  FaArrowDownRegular,
+  FaArrowRightRegular,
+  FaArrowUpRegular,
+  FaCircleCheckRegular,
+} from '@origin/shared/icons';
+import { format } from 'dnum';
+
+import { wagmiConfig } from '../../clients';
+
+import type { ReactNode } from 'react';
+import type { IntlShape } from 'react-intl';
+
+import type {
+  Activity,
+  ActivityType,
+  ApprovalActivity,
+  BridgeActivity,
+  ClaimRewardsActivity,
+  DelegateVoteActivity,
+  ExtendStakeActivity,
+  MigrateActivity,
+  RedeemActivity,
+  StakeActivity,
+  SwapActivity,
+  UnstakeActivity,
+  VoteActivity,
+} from './types';
+
+export type ActivityOption = {
+  title: (activity: Activity, intl: IntlShape) => string;
+  subtitle: (activity: Activity, intl: IntlShape) => string;
+  icon: (activity: Activity) => ReactNode;
+};
+
+export const activityOptions: Record<ActivityType, ActivityOption> = {
```

Comment:
I see you needed this so you could pull the behavior into your pushNotification actions.

Reply:
the idea here is more about splitting presentation and content, this constant file exposes all content for all kind af activities while it is later consumed by the `ActivityTile` or the `NotificationSnack` for presentation. Since we need the activity parameters in a serializable way, these options are making the mapping between args and output.
This pattern also scales quite well as we can add any required information to the option and add as many activity type to this list.